.github/workflows/cluster_endtoend_onlineddl_fk.yml

```diff
@@ -0,0 +1,141 @@
+# DO NOT MODIFY: THIS FILE IS GENERATED USING "make generate_ci_workflows"
+
+name: Cluster (onlineddl_fk)
+on: [push, pull_request]
+concurrency:
+  group: format('{0}-{1}', ${{ github.ref }}, 'Cluster (onlineddl_fk)')
+  cancel-in-progress: true
```

Comment:
Honest questions:

1. if the unit test uses an actual `mysqld` server, why is it called a unit test? It sounds like it's an `endtoend` test in disguise. 
2. Other than consolidating the tests in fewer workflow files, what is the advantage of moving this into a unit test?

Reply:
But let me look into consolidating this test into `onlineddl_vrepl`, where I think it's appropriate.

Reply:
consolidating into `onlineddl_vrepl` was not a good idea, because `onlineddl_vrepl` uses a sharded schema, and foreign key experimentation does not include sharded environments yet.

Reply:
Good news! `onlineddl_fk` tests consolidated into `onlineddl_scheduler`, and there are no new workflows in this PR.

Reply:
It gets a little blurry when unit tests have external processes, I agree. In some cases, however, you cannot really test the vitess software component w/o a mysqld and OnlineDDL is a good example I think. The test binary with 1 vttestserver and 1 mysqld becomes the smallest unit. In practical terms, the test will generally run the same locally as it does in the CI (things like VT_MYSQL_ROOT differing of course) and in the CI it will complete in say 30 seconds vs 10 minutes.

I understand the push back though, I was exactly the same way. I resisted relying more on the unit tests for quite a while but I'm glad I am using them more now. 

Glad that you were able to consolidate these ❤️ ! The biggest chunk of time in the CI seems to be in the workflow setup so limiting the number of workflows has clear value IMO.

Reply:
OK cool, let me first see if I can consolidate even more tests (I'm thinking `onlineddl_singleton` would be a good candidate). Then, at some point, perhaps we can pair and you can show me how `vttestserver` works, and see how it goes?

Reply:
Sounds good!

Reply:
Followup: https://github.com/vitessio/vitess/pull/12055